### PR TITLE
close: Print multiple txs; Fixes #6467

### LINKED ISSUE #6467: close() should print multiple txs and txids if we close during a splice.
Needs a schema change, probably new "txids" and "txs" fields?

### PR DESCRIPTION
**Pull request by ddustin**
`close` now outputs txs & txids of all closing transactions (splice candidates can cause there to be multiple).

### CODE REVIEW

File: lightningd/closing_control.c [hunk 1]
```diff
@@ -52,16 +52,36 @@ struct close_command {
...
 {
+	assert(tal_count(close_txs));
 	struct json_stream *result = json_stream_success(cc->cmd);
+	const struct bitcoin_tx *close_tx = close_txs[tal_count(close_txs) - 1];
 
 	json_add_tx(result, "tx", close_tx);
 	if (!invalid_last_tx(close_tx)) {
 		struct bitcoin_txid txid;
 		bitcoin_txid(close_tx, &txid);
 		json_add_txid(result, "txid", &txid);
```
Comment (niftynei) [inline lightningd/closing_control.c:65]: nit: deprecate the `txid` field, in favor of just using the `txids` field in the future.

File: lightningd/closing_control.c [hunk 2]
```diff
@@ -52,16 +52,36 @@ struct close_command {
...
 	}
+
+	json_array_start(result, "txs");
+	for (int i = 0; i < tal_count(close_txs); i++)
+		json_add_tx(result, NULL, close_txs[i]);
+	json_array_end(result);
+
+	json_array_start(result, "txids");
+	for (int i = 0; i < tal_count(close_txs); i++) {
+		if (invalid_last_tx(close_txs[i])) {
```
Comment (niftynei) [inline lightningd/closing_control.c:75]: Man this is an old bugfix (v0.7.0)

**Review by niftynei — COMMENTED** [review]
Can you deprecate the `txid` field in `close`?

**Comment by niftynei** [comment]
the test failure is from one of the new fetchinvoice/offers tests

cc @rustyrussell 
[logs_25912008146_fetchinvoice.zip](https://github.com/user-attachments/files/16238661/logs_25912008146_fetchinvoice.zip)

File: lightningd/closing_control.c (hunk 1 above)
Comment (ddustin) [reply lightningd/closing_control.c:65]: I deprecated `txid` and `tx`

Comment (niftynei) [reply lightningd/closing_control.c:65]: ah ok, missed it the first time! seems like the deprecation feature stuff got updated, very cool.

**Review by niftynei — COMMENTED** [review]
ACK 778aae88792beb4c04c4a7addc91776b1c9f73e2

**Comment by rustyrussell** [comment]
Rebased, but also deprecated properly, see the second commit.

1. Deprecated field in the schema now has two values, start and end.
2. We actually test deprecation status at runtime.
3. We document all deprecations.

You'll note this broke tests, too.  This is important; if we break an API we should absolutely feel the pain ourselves!

**Comment by rustyrussell** [comment]
Trivial rebase.

**Comment by rustyrussell** [comment]
Rebased again, and changed deprecation from 24.08 to 24.11...

**Pull request merged**